baseplate/sidecars/event_publisher.py

```diff
@@ -100,14 +104,15 @@ def serialize(self) -> SerializedBatch:
 
 class BatchPublisher:
     def __init__(self, metrics_client: metrics.Client, cfg: Any):
+        bp = Baseplate()
+        bp.configure_context({
+            "http_client": ExternalRequestsClient("event_collector"),
+        })
+        self.baseplate = bp
         self.metrics = metrics_client
         self.url = f"{cfg.collector.scheme}://{cfg.collector.hostname}/v{cfg.collector.version}"
         self.key_name = cfg.key.name
         self.key_secret = cfg.key.secret
-        self.session = requests.Session()
-        self.session.headers[
-            "User-Agent"
-        ] = f"baseplate.py-{self.__class__.__name__}/{baseplate_version}"
```

Comment:
Maybe I'm missing something, but it seems like this was getting overwritten below anyway